File: mobile/lib/services/arvr_service.dart
import 'package:http/http.dart' as http;
import 'dart:convert';

class ARVRService {
  final String apiBaseUrl;

  ARVRService(this.apiBaseUrl);

  Future<dynamic> generateARVRContent(String courseId) async {
    final response = await http.post(
      Uri.parse('$apiBaseUrl/arvr/generate'),
      headers: {'Content-Type': 'application/json'},
      body: json.encode({'courseId': courseId}),
    );
    return json.decode(response.body);
  }
}

File: frontend/src/services/arvrService.js
import api from './api';

export const generateARVRContent = async (courseId) => {
  try {
    const response = await api.post('/arvr/generate', { courseId });
    return response.data;
  } catch (error) {
    console.error('Failed to generate AR/VR content:', error);
    throw new Error(error.response?.data?.message || 'Failed to generate content');
  }
};
